sysstat.service.in

```diff
@@ -2,20 +2,27 @@
 # (C) 2012 Peter Schiffer (pschiffe <at> redhat.com)
 #
 # @PACKAGE_NAME@-@PACKAGE_VERSION@ systemd unit file:
-#	 Insert a dummy record in current daily data file.
-#	 This indicates that the counters have restarted from 0.
+#     1. Insert a dummy record in current daily data file when starting/stopping
+#        the unit. This indicates that the counters have restarted from 0.
+#     2. This unit is automatically stopped/started on suspend/resume,
+#        see /usr/lib/systemd/system-sleep/sysstat-sleep.sh.
+#     3. Starting this unit also starts the timer unit for periodic accounting.
 
 [Unit]
-Description=Resets System Activity Logs
+Description=System Activity Accounting
+# Start the periodic accounting process
+Requires=sysstat-collect.timer
+# but make sure we finish first, because the accounting process locks the datafile
+Before=sysstat-collect.timer sysstat-collect.service
 
 [Service]
 Type=oneshot
 RemainAfterExit=yes
 User=@CRON_OWNER@
-ExecStart=@SA_LIB_DIR@/sa1 --boot
+ExecStart=@SA_LIB_DIR@/sa1 --boot -C STARTED
```

Comment:
Hmm it was on purpose that I removed the "LINUX-RESTART" line, since I thought it was only serving as a comment, and that it didn't affect functionality. So I replaced a comment with a more accurate one, since resuming is not really a "linux-restart". FYI having a comment at the suspend-resume and shutdown-restart points, helps when you are plotting the data, to insert a gap in the graph.

When you say the LINUX-RESTART line resets the counters, how does this affect sysstat's functionality, and which components of sysstat?

I'm thinking that if it changes the logging behaviour, then I can't put that command back because it will cause the counter-reset on every resume. Which means that I'll have to think another way of writing this record on suspend/resume.